### PR TITLE
Updates the PX4 Architecture diagrams.

### PR DESCRIPTION
**Pull request by hamishwillee**
This updates the Dronecode Platform architecture doc. There is no "dronecode platform" in the way that was presented. The change splits the image into two cases - with FC and FC+ mission computer. It attempts to only present things that are "supported" - ie it isn't aspirational.

@bkueng Can you please review. If you are broadly happy with it I will further
- Link up as many bits as I can to existing docs
- rename the actual source file.

FYI @LorenzMeier

### CODE REVIEW

**Review by LorenzMeier — APPROVED** [review]

File: en/concept/dronecode_architecture.md [hunk 1]
```diff
@@ -1,17 +1,53 @@
...
+
+The diagram below provides a high level overview of a typical "simple" PX4 system based around a flight controller.
+
+![PX4 architecture - FC only system](../../assets/diagrams/px4_arch_fc.svg)
+
+<!-- Source for drawing: https://docs.google.com/drawings/d/1_2n43WrbkWTs1kz0w0avVEeebJbfTj5SSqvCmvSOBdU/edit -->
+
+The hardware consists of 
+- [Flight controller](../flight_controller/README.md) (running the PX4 flight stack). This often includes internal IMUs, compass and barometer.
+- [Motor ESCs](../peripherals/esc_motors.md) connected to [PWM outputs](../peripherals/pwm_escs_and_servo.md) or a [UAVCAN](../peripherals/uavcan_escs.md) bus (UAVCAN allows two-way communication, not single direction as shwown).
```
Comment (bkueng) [inline en/concept/dronecode_architecture.md:21]: I'd keep this a bit broader as there's no only the 2 protocols, or motors: Actuators (motors and servos).
Also on the diagram

File: en/concept/dronecode_architecture.md [hunk 2]
```diff
@@ -1,17 +1,53 @@
...
+- The ground station computer typically runs QGroundControl (or some other ground station software).
+  It may also run robotics software like MAVSDK or ROS.
+- The PX4 flight stack running on the flight controller includes drivers, comms and other middleware and numerous modules.
+
+
+## FC and Mission Computer
+
+The diagram below shows a PX4 system that includes both a flight controller and a mission computer.
+
+![PX4 architecture - FC only system](../../assets/diagrams/px4_arch_fc_companion.svg)
```
Comment (bkueng) [inline en/concept/dronecode_architecture.md:37]:
```suggestion
![PX4 architecture - FC and Mission Computer](../../assets/diagrams/px4_arch_fc_companion.svg)
```

**Review by bkueng — COMMENTED** [review]
Yes looks good

**Comment by hamishwillee** [comment]
Thanks for review. Not sure why it is failing, but I will check , fix issue, and merge (probably tomorrow)

File: en/concept/dronecode_architecture.md (hunk 1 above)
Comment (hamishwillee) [reply en/concept/dronecode_architecture.md:21]: OK, so add Dshot and also say "or other protocols"?

Comment (bkueng) [reply en/concept/dronecode_architecture.md:21]: Yes, but there's also more buses, like I2C or serial.

Comment (hamishwillee) [reply en/concept/dronecode_architecture.md:21]: Ah, I'd only seen ESC connected to PWM or UAVCAN - had assumed I2C or serial were never used for this

Comment (hamishwillee) [reply en/concept/dronecode_architecture.md]: Fixed.

**Comment by hamishwillee** [comment]
I've fixed up following feedback, moved the page, cross linked to relevant resources. Now merging. @bkueng I think this is good, but feel free to review again!

**Pull request merged**